pymeasure/instruments/keithley/keithley2400.py

```diff
@@ -634,128 +1111,121 @@ def output_trigger_on_external(self, line=1, after="DEL"):
-        currRange = 1.2 * max(abs(stopI), abs(startI))
-        # self.write(":SOUR:CURR 0.0")
-        self.write(":SENS:VOLT:PROT %g" % compliance)
-        self.write(":SOUR:DEL %g" % delay)
-        self.write(":SOUR:CURR:RANG %g" % currRange)
-        self.write(":SOUR:SWE:RANG FIX")
-        self.write(":SOUR:CURR:MODE SWE")
-        self.write(":SOUR:SWE:SPAC LIN")
-        self.write(":SOUR:CURR:STAR %g" % startI)
-        self.write(":SOUR:CURR:STOP %g" % stopI)
-        self.write(":SOUR:CURR:STEP %g" % stepI)
-        self.write(":TRIG:COUN %d" % num)
-        if backward:
-            currents = np.linspace(stopI, startI, num)
-            self.write(":SOUR:SWE:DIR DOWN")
-        else:
-            currents = np.linspace(startI, stopI, num)
-            self.write(":SOUR:SWE:DIR UP")
-        self.connection.timeout = 30.0
-        self.enable_source()
-        data = self.values(":READ?")
+    line_frequency_auto = Instrument.control(
+        ":SYST:LFR:AUTO?",
+        ":SYST:LFR:AUTO %d",
+        """ Control the auto line frequency (boolean).
 
-        self.check_errors()
-        return zip(currents, data)
+        .. deprecated:: 0.16
+           Use :attr:`~.auto_zero_enabled` or :meth:`~.auto_zero_once`.""",
+        values={True: 1, False: 0},
+        map_values=True,
+        get_process=_deprecate_process(
+            "Deprecated to use `Keithley2400.line_frequency_auto`, "
+            "use `Keithley2400.line_frequency_auto_enabled`",
+        ),
+        set_process=_deprecate_process(
+            "Deprecated to use `Keithley2400.line_frequency_auto`, "
+            "use `Keithley2400.line_frequency_auto_enabled`",
+        ),
+    )
 
-    def RvsIaboutZero(self, minI, maxI, stepI, compliance, delay=10.0e-3):
-        data = []
-        data.extend(self.RvsI(minI, maxI, stepI, compliance=compliance, delay=delay))
-        data.extend(self.RvsI(minI, maxI, stepI, compliance=compliance, delay=delay, backward=True))
-        self.disable_source()
-        data.extend(self.RvsI(-minI, -maxI, -stepI, compliance=compliance, delay=delay))
-        data.extend(
-            self.RvsI(-minI, -maxI, -stepI, compliance=compliance, delay=delay, backward=True)
-        )
-        self.disable_source()
-        return data
+    front_terminals_enabled = Instrument.control(
+        ":ROUTE:TERMINALS?",
+        ":ROUTE:TERMINALS %s",
+        """Control whether the to route to the front terminals (bool).
+        When True the front terminals are routed to and the rear disconnected,
+        and vice-versa when False.""",
+        validator=strict_discrete_set,
+        values={True: "FRON", False: "REAR"},
+        map_values=True,
+    )
 
-    def use_rear_terminals(self):
-        """Enable the rear terminals for measurement, and
-        disable the front terminals."""
-        self.write(":ROUT:TERM REAR")
+    def use_rear_terminals(self):  # Included for backwards compatibility
```

Comment:
Instead of the comment, please add a deprecation note and warning, such that we can remove these methods one day.